compiler/rustc_hir_typeck/src/fallback.rs

```diff
@@ -53,34 +49,32 @@ impl<'tcx> FnCtxt<'_, 'tcx> {
 
         let fallback_occurred = self.fallback_types();
 
-        if !fallback_occurred {
-            return;
+        if fallback_occurred {
+            // We now see if we can make progress. This might cause us to
+            // unify inference variables for opaque types, since we may
+            // have unified some other type variables during the first
+            // phase of fallback. This means that we only replace
+            // inference variables with their underlying opaque types as a
+            // last resort.
```

Comment:
existing, but I do not understand that comment?

Do you understand what this wants to say. It feels like this is a general "if fallback occurred, previously stalled goals may make progress again".

I really don't see how there's anything specific to opaques here.